### ISSUE TITLE
Mock an API response to fail on the first try and succeed on the second attempt. Is it achievable with Requestly?

### ISSUE THREAD

**Issue by nafees87n** [requestly/requestly#1938]
### [modify response rule](https://developers.requestly.com/http-rules/modify-response-body/) along with window variables can be used to keep track of request counts.

![image](https://github.com/requestly/requestly/assets/56021937/c46ecaae-9144-4220-b110-3ac7e946cc88)

1.  Javascript mode of modify response rule can be used.
2. `window._requestCount` keeps count on the number of requests based on which responses could be returned.

Please find the sharedlist [here](https://app.requestly.io/rules#sharedList/1720706365592-request-fail-on-first-try). This rule can be used on https://testheaders.com/

**Comment by wrongsahil** [comment]
I don't think this approach will be able to return different response`Status Code` 🤔. But yeah, works for response body.

**Comment by Ronakkadhi** [comment]
Closing as stale — no activity for over a year.

As part of the 2026 repository restructure, we're closing dormant issues to focus on actively-tracked items. If this is still relevant, please re-open or file a fresh issue at the right home:

- 🌐 **HTTP Interceptor** (Chrome extension + web app + desktop proxy) → [requestly/interceptor](https://github.com/requestly/interceptor/issues/new/choose)
- 🚀 **Requestly API Client** → this repo

Thanks for taking the time to file the original report.
